Assets/Scripts/Combat/Abilities/Bonk.cs

```diff
@@ -0,0 +1,170 @@
+using Assets.Scripts.Combat;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace Combat.Abilities
+{
+    public class Bonk : Ability
+    {
+
+        private enum Phase { Inactive, Approaching, Bonking, Disengaging }
+        private Phase CurrentPhase = Phase.Inactive;
+
+        private Timer Timer;
+        private const float ApproachingDuration = 2.5f;
+        private const float BonkingDuration = 3.0f;
+        private const float DisengagingDuration = 1.5f;
+
+        private const float Damage = 25;
+
+        private Vector3 InitialPosition;
+        private GameObject Victim;
+        private Vector3 AttackingPosition;
+        private readonly float SpaceBetweenBonk = 2.5f;
+
+        // Maybe there's a better way of finding this out dynamically...?
+        [SerializeField]
+        private bool BelongsToAlly;
```

Comment:
Enemies don't have costumes right? If you can check that whatever owns this ability doesn't have a Costume, you could get this value dynamically. Another way would be to check if the owner has an EnemyCombatant component or something similar.

Reply:
I'd prefer to check for the Enemy / Ally Combatant component, since that seems more accurate. Thanks!! I'll add this in and remove the boolean param

Reply:
Updated!